spec/features/date_selection_spec.rb

```diff
@@ -0,0 +1,195 @@
+RSpec.describe 'date selection', type: :feature do
+  include GdsApi::TestHelpers::ContentDataApi
+
+  before do
+    initial_page_stub
+  end
+
+  it 'renders data for the last 30 days if no date period is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 30.days
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+
+    expect_default_date_period_metrics_to_be_displayed(from, to)
+  end
+
+  it 'renders data for the last 30 days when `last 30 days` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 30.days
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-30-days').click
+    click_button 'OK'
+    sleep 2
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  it 'renders data for the previous month when `last month` is selected' do
+    to = Time.zone.today.last_month.end_of_month
+    from = Time.zone.today.last_month.beginning_of_month
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
```

Comment:
You could extract a method for lines 35-39 like:

```ruby
def when_i_filter_by_date_range('last-month')
#
#
#
end
```

Reply:
Done! Also added `visit ...` to this method.